libs/blocks/merch-card/merch-card.js

```diff
@@ -1,184 +1,188 @@
+  }
+  const actionMenuContentWrapper = el.children[index];
+  const actionMenuContent = actionMenuContentWrapper.children[0];
+  actionMenuContentWrapper.remove();
+  return actionMenuContent;
 };
 
+function getMerchCardRows(rows, ribbonMetadata, cardType, actionMenuContent) {
+  if (cardType === 'catalog') {
+    const index = ribbonMetadata === null ? 0 : 1;
+    return actionMenuContent !== null ? rows[index + 1] : rows[index];
+  }
+  return rows[ribbonMetadata === null ? 0 : 1];
+}
+
 const init = (el) => {
-  loadStyle(`${base}/deps/caas.css`);
-  const section = el.closest('.section');
-  section.classList.add('milo-card-section');
-  const images = el.querySelectorAll('picture');
-  let image;
-  const icons = [];
-  const rows = el.querySelectorAll(':scope > *');
-  const ribbonMetadata = rows[0].children?.length === 2 ? rows[0].children : null;
-  const row = rows[ribbonMetadata === null ? 0 : 1];
-  const altCta = rows[rows.length - 1].children?.length === 2
-    ? rows[rows.length - 1].children : null;
-  const allPElems = row.querySelectorAll('p');
-  const ctas = allPElems[allPElems.length - 1];
-  const styles = [...el.classList];
-  const cardType = getPodType(styles);
-  const merchCard = el;
-  decorateBlockHrs(el);
-  images.forEach((img) => {
-    const imgNode = img.querySelector('img');
-    const { width, height } = imgNode;
-    const isSquare = Math.abs(width - height) <= 10;
-    if (img) {
-      if (isSquare) {
-        icons.push(img);
-      } else {
-        image = img;
+  let section = el.closest('.section');
+  const upClass = getUpFromSectionMetadata(section);
+  if (upClass) {
+    import('./legacy-merch-card.js')
+      .then((module) => {
+        module.initLegacyMerchCard(el);
+      });
+  } else {
+    if (section.parentElement.classList.contains('fragment')) {
+      const fragment = section.parentElement;
+      const fragmentParent = fragment.parentElement;
+      section.style.display = 'contents';
+      fragment.style.display = 'contents';
+      fragmentParent.style.display = 'contents';
+      section = fragmentParent.parentElement;
+    }
+    const headings = el.querySelectorAll('h1, h2, h3, h4, h5, h6');
+    decorateLinkAnalytics(el, headings);
+    const images = el.querySelectorAll('picture');
+    let image;
+    const icons = [];
+    const rows = el.querySelectorAll(':scope > *');
+    const styles = [...el.classList];
+    const cardType = getPodType(styles);
+    const ribbonMetadata = rows[0].children?.length === 2 ? rows[0].children : null;
+    const actionMenuContent = cardType === 'catalog' ? getActionMenuContent(el, ribbonMetadata) : null;
+    const row = getMerchCardRows(rows, ribbonMetadata, cardType, actionMenuContent);
+    const altCta = rows[rows.length - 1].children?.length === 2
+      ? rows[rows.length - 1].children : null;
+    const allPElems = row.querySelectorAll('p');
+    const ctas = allPElems[allPElems.length - 1];
+    images.forEach((img) => {
```

Comment:
We'll probably optimize this in a subsequent PR once this is merged to fetch an image from Fastly that is optimized for the merch-card specs. Currently, it uses the original image that an author has added which might be over-dimensioned and potentially unnecessarily large in size.